subprojects/core/src/integTest/groovy/org/gradle/api/internal/initialization/loadercache/ClassLoadersCachingIntegrationTest.groovy

```diff
@@ -418,4 +418,30 @@ class ClassLoadersCachingIntegrationTest extends PersistentBuildProcessIntegrati
         isCached("a")
         isNotCached("a:a")
     }
+
+    @ToBeFixedForConfigurationCache(because = "test relies on static state")
```

Comment:
I guess we won't fix this test to work with the configuration cache so `@UnsupportedWithConfigurationCache` would better.

Reply:
@bamboo same goes for all the tests in this class. The behaviour should be consistent when using config caching though, so either these tests need to be updated to be compatible or the config caching specific tests generalised so that these can be removed.

Reply:
Can't the test use static state when configuration caching is enabled? The ClassLoaders should be cached in the same way regardless of whether it is enabled or not (assuming a type defined in the scripts is referenced by the task graph).

Reply:
> assuming a type defined in the scripts is referenced by the task graph

The static state isn't, as it is currently written.